tuf/signed/verify.go

```diff
@@ -77,10 +77,11 @@ func VerifySignatures(s *data.Signed, roleData data.BaseRole) error {
 		if key.ID() != sig.KeyID {
 			return ErrInvalidKeyID{}
 		}
-		if err := VerifySignature(msg, sig, key); err != nil {
+		if err := VerifySignature(msg, &sig, key); err != nil {
 			logrus.Debugf("continuing b/c %s", err.Error())
 			continue
 		}
+		s.Signatures[i].IsValid = sig.IsValid
```

Comment:
Maybe we can just set `IsValid` to true, since `VerifySignature` will error if the signature is not valid, and the line should never be reached if it's invalid?  That way `VerifySignature` mutates nothing, and only `VerifySignatures` actually does the mutation.